src/Display/Collapsible/Collapsible.tsx

```diff
@@ -9,72 +10,63 @@ import {
   CollapsibleBody,
 } from '../../Style/Display/CollapsibleStyle';
 
-class Collapsible extends React.Component<Props, State> {
-  state = {
-    isOpen: true,
-  };
+const Collapsible: React.FunctionComponent<Props> = ({
+  label,
+  children,
+  className,
+  isOpen,
+  ...defaultProps
+}) => {
+  // If controlled, set initial state to isOpen prop
+  // If uncontrolled, set initial state to open
+  const [isOpenInternal, setIsOpenInternal] = React.useState<boolean>(
+    isUndefined(isOpen) ? true : isOpen
+  );
 
-  handleOpen = () => {
-    const { isOpen } = this.state;
-    this.setState({ isOpen: !isOpen });
-  };
-
-  componentDidMount() {
-    const { isOpen } = this.props;
-
-    if (isOpen === false && isOpen !== undefined) {
-      this.setState({
-        isOpen,
-      });
+  // If controlled, don't change the state on click
+  // If uncontrolled, toggle change
+  const toggle = React.useCallback(() => {
```

Comment:
I think `useCallback` is not needed here since it is passed to a div.

Reply:
What's the difference? I'm using `useCallback` here mostly because I'm defaulting to that instead of creating a plain function (which would get recreated on every render). But tbh while I don't know which has better performance, I'd guess it will never make a difference.

Reply:
I usually use `useCallback` for `PureComponent` or the dependency of hooks to prevent unnecessary render. 

But in this case, I think `CollapsibleContainer` will re-render every time because it's a node rather than a component. Therefore, the `useCallback` will be the extra cost.